### PR TITLE
Add RTKQuery Fetch Sessions query

### PR DESCRIPTION
**Pull request by brianwachira**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

- Adds feature to save user on Async storage and state
- Adds RTKQuery fetch sessions query feature.

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/services/auth.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        }
+    }),
+    getSchedule: builder.query<Schedule, void>({
+        query: () => {
+          return {
+            url: `/events/${EVENT_SLUG}/schedule?grouped=true`,
+            method: 'GET',
+          }
+        }
+    })
```
Comment (makunomark) [inline src/services/auth.ts:61]: Based on the name of the file, seems like these two apis `getSchedule` and `getSessions` may be misplaced.

**Review by makunomark — APPROVED** [review]
LGTM

File: src/services/auth.ts (hunk 1 above)
Comment (brianwachira) [reply src/services/auth.ts:61]: Lemme remove them ASAP

Comment (brianwachira) [reply src/services/auth.ts:61]: I have removed it

**Pull request merged**